src/main/java/com/xpanxion/java/springboot/da1/demo/service/student2/HistoryService2.java

```diff
@@ -1,33 +1,56 @@
 package com.xpanxion.java.springboot.da1.demo.service.student2;
 
-import com.xpanxion.java.springboot.da1.demo.model.student2.History2;
-import com.xpanxion.java.springboot.da1.demo.repository.student2.HistoryRepository2;
+import com.xpanxion.java.springboot.da1.demo.model.student2.Checkin2;
+import com.xpanxion.java.springboot.da1.demo.model.student2.Checkout2;
+import com.xpanxion.java.springboot.da1.demo.repository.student2.CheckinRepository2;
+import com.xpanxion.java.springboot.da1.demo.repository.student2.CheckoutRepository2;
 import com.xpanxion.java.springboot.da1.demo.repository.student2.MemberRepository2;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
 
 import java.sql.Timestamp;
 
 @Service
 public class HistoryService2 {
 
-    History2 history = new History2();
+    Checkin2 checkin = new Checkin2();
+    Checkout2 checkout = new Checkout2();
 
     @Autowired
     private MemberRepository2 memberRepository2;
 
     @Autowired
-    private HistoryRepository2 historyRepository2;
+    private CheckinRepository2 checkinRepository2;
 
-    public History2 addHistoryIn(int memberId, Timestamp time) {
-        history.setCheckin(time);
-        history.setMember2(memberRepository2.findById(memberId));
-        return historyRepository2.save(history);
+    @Autowired
+    private CheckoutRepository2 checkoutRepository2;
+
+    public Checkin2 addHistoryIn(int memberId, Timestamp time) {
+        checkin.setCheckin(time);
+
+        try {
+            checkin.setMember2(memberRepository2.findById(memberId));
+            if (checkin.getMember2() == null) {
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "member id not found");
+            }
+        } catch (Exception e) {
+            throw e;
```

Comment:
Close. :-)  We don't need to catch the Exception and re-throw it.  We can remove the try/catch blocks and simply throw a new `ResponseStatusException`.  Can you make these changes?